mpt_trie/examples/simple.rs

```diff
@@ -44,7 +44,7 @@ fn main() {
 
     // Values can be deleted:
     let del_val = trie.delete(0x1234_u32);
-    assert_eq!(del_val, Some(b"some data".to_vec()));
+    assert_eq!(del_val.ok().unwrap(), Some(b"some data".to_vec()));
```

Comment:
I'm not convinced I see the benefit of `unwrap()` here...

Reply:
what do you mean?

Reply:
Given the code is in examples we could do something like

```
let value = trie.delete(0x1234_u32)?.expect("well that went wrong for whatever reason");
assert_eq!(value, b"some data".to_vec());
assert_eq!(trie.get(0x1234_u32), None);
```
which is easier to parse to me. Propagate the error up, expect on None, and move to the assert if we're dandy.

We could even do something like

```
let value = trie.delete(0x1234_u32)?.ok_or(BigTrieError::SuperBadError)?;
assert_eq!(value, b"some data".to_vec());
assert_eq!(trie.get(0x1234_u32), None);
```

but that's probably overkill for here

Reply:
got it, makes sence